### DISCUSSION TITLE
Proposals For Critical Fixes

### DISCUSSION THREAD

**Discussion by dip000** [dip000/godot-landscaper#4] [Announcements]
Comment in this thread any crítical fixes that need to be patched up before launching to the Godot Asset Store. Issues are closed (because Idk how to do that 😃)

**Comment by mohdalsalmi** [comment]
Hello, 
I just downloaded the plugin to try it. It worked as you said, but i just have two problems:

1: no undo/redo 
2: no shadows can be casted on the terrain

**Comment by jinyangcruise** [comment]
Thanks for this awesome plugin!
I have a suggestion: let users control the grass rotation direction and degree:

In res://addons/godot_landscaper/scripts/brush_grass_spawn.gd:
```
...
var random_rotation:float = _rng.randf_range( 0, 2*PI )
...
transf = transf.rotated_local( Vector3.UP, random_rotation )
...
```
Change this to something like:
```
...
var random_rotation_up:float = _rng.randf_range( 0, custom_rotation_up ) # User customized
var random_rotation_right:float = _rng.randf_range( 0, custom_rotation_right ) # User customized
var random_rotation_back:float = _rng.randf_range( 0, custom_rotation_back ) # User customized
...
# rotation order is YXZ
transf = transf.rotated_local( Vector3.UP, random_rotation_up )
transf = transf.rotated_local( Vector3.RIGHT, random_rotation_right )
transf = transf.rotated_local( Vector3.BACK, random_rotation_back )
...
```

**Reply by dip000** [reply]
Yeah, that'd be cool.
I'll update and refactor a few things on the way when i can. Thanks!

**Comment by dip000** [comment]
Welp, this is awkward.
I've found myself starting a game dev journey, but the current main version is TERRIBLY unsafe to use for a more serious development!

Anyway, I'm rebuilding this stuff. I've got a grass spawner-colorer with quads, multimesh instances, and color baking. The undo-redo, and custom rotations and scales as well.

Humm, thanks for tuning in. [See ya in v0.3](https://github.com/dip000/godot-landscaper/tree/v0.3)!
